Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -0,0 +1,264 @@
+  public func configure(with data: PledgeViewData) {
+    self.configureWithDataProperty.value = data
+  }
+
+  private let (pledgeAmountDataSignal, pledgeAmountObserver) = Signal<PledgeAmountData, Never>.pipe()
+  public func pledgeAmountViewControllerDidUpdate(with data: PledgeAmountData) {
+    self.pledgeAmountObserver.send(value: data)
+  }
+
+  private let (shippingRuleSelectedSignal, shippingRuleSelectedObserver) = Signal<ShippingRule, Never>.pipe()
+  public func shippingRuleSelected(_ shippingRule: ShippingRule) {
+    self.shippingRuleSelectedObserver.send(value: shippingRule)
+  }
+
+  private let (submitButtonTappedSignal, submitButtonTappedObserver) = Signal<Void, Never>.pipe()
+  public func submitButtonTapped() {
+    self.submitButtonTappedObserver.send(value: ())
+  }
+
+  private let viewDidLoadProperty = MutableProperty(())
+  public func viewDidLoad() {
+    self.viewDidLoadProperty.value = ()
+  }
+
+  // MARK: - Outputs
+
+  public let configureLocalPickupViewWithData: Signal<PledgeLocalPickupViewData, Never>
+  public let configurePledgeAmountViewWithData: Signal<PledgeAmountViewConfigData, Never>
+  public let configureShippingLocationViewWithData: Signal<PledgeShippingLocationViewData, Never>
+  public let configureShippingSummaryViewWithData: Signal<PledgeShippingSummaryViewData, Never>
+  public let localPickupViewHidden: Signal<Bool, Never>
+  public let pledgeAmountViewHidden: Signal<Bool, Never>
+  public let shippingLocationViewHidden: Signal<Bool, Never>
+  public let shippingSummaryViewHidden: Signal<Bool, Never>
+
+  public var inputs: ConfirmDetailsViewModelInputs { return self }
+  public var outputs: ConfirmDetailsViewModelOutputs { return self }
+}
+
+// MARK: - Helper Functions
+
+private func pledgeSummaryViewData(
+  project: Project,
+  total: Double,
+  confirmationLabelHidden: Bool
+) -> PledgeSummaryViewData {
+  return (project, total, confirmationLabelHidden)
+}
+
+private func pledgeAmountSummaryViewData(
+  with project: Project,
+  reward _: Reward,
+  allRewardsTotal: Double,
+  additionalPledgeAmount: Double,
+  shippingViewsHidden: Bool,
+  context: PledgeViewContext
+) -> PledgeAmountSummaryViewData? {
+  guard let backing = project.personalization.backing else { return nil }
+
+  let rewardIsLocalPickup = isRewardLocalPickup(backing.reward)
+  let projectCurrencyCountry = projectCountry(forCurrency: project.stats.currency) ?? project.country
+
+  return .init(
+    bonusAmount: additionalPledgeAmount,
+    bonusAmountHidden: context == .update,
+    isNoReward: backing.reward?.isNoReward ?? false,
+    locationName: backing.locationName,
+    omitUSCurrencyCode: project.stats.omitUSCurrencyCode,
+    projectCurrencyCountry: projectCurrencyCountry,
+    pledgedOn: backing.pledgedAt,
+    rewardMinimum: allRewardsTotal,
+    shippingAmount: backing.shippingAmount.flatMap(Double.init),
+    shippingAmountHidden: !shippingViewsHidden,
+    rewardIsLocalPickup: rewardIsLocalPickup
+  )
+}
```

Comment:
These are actually for the next pieces. Forgot to remove these for this initial PR